packages/react-celo/src/connectors/injected.ts

```diff
@@ -33,23 +33,28 @@ export default class InjectedConnector
     this.network = network;
   }
 
-  async initialise(): Promise<this> {
+  async initialise(lastUsedAddress?: string): Promise<this> {
     if (this.initialised) {
       return this;
     }
 
+    let defaultAccount = lastUsedAddress;
+
     const injected = await getInjectedEthereum();
     if (!injected) {
       throw new Error('Ethereum wallet not installed');
     }
     const { web3, ethereum, isMetaMask } = injected;
 
     this.type = isMetaMask ? WalletTypes.MetaMask : WalletTypes.Injected;
-
-    const [defaultAccount] = await ethereum.request({
-      method: 'eth_requestAccounts',
-    });
-
+    const metamask = ethereum._metamask;
+    const isUnlocked = isMetaMask && (await metamask?.isUnlocked());
+    const isConnected = ethereum.isConnected && ethereum.isConnected();
+    if (isUnlocked || !isConnected || !lastUsedAddress) {
```

Comment:
If you change this to use `defaultAccount` instead of `lastUsedAddress` you can get rid of the `as string` casting below.